### PR TITLE
make emagged marker component

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
adds `EmaggedComponent` marker to control emagging (prevent emagging a thing twice, unless you really want to)

will eventually replace the ~8 things that use Emagged/IsEmagged/!Safe as component fields

**Media**
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
no

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
You might as well just implement it. Emag state isn't stored in that many places.

File: Content.Shared/VendingMachines/SharedVendingMachineSystem.cs [hunk 1]
```diff
@@ -1,10 +1,12 @@
+using Content.Shared.Emag.Systems;
 using Robust.Shared.Prototypes;
 using System.Linq;
 
 namespace Content.Shared.VendingMachines;
 
 public abstract class SharedVendingMachineSystem : EntitySystem
 {
+    [Dependency] protected readonly EmagSystem _emagSystem = default!;
```
Comment (deltanedas) [inline Content.Shared/VendingMachines/SharedVendingMachineSystem.cs:9]: is making it protected ok or should this be private + have another in server VendingMachineSystem

File: Content.Server/Access/AccessWireAction.cs [hunk 2]
```diff
@@ -1,39 +1,43 @@
...
 using Content.Shared.Access.Components;
+using Content.Shared.Emag.Components;
+using Content.Shared.Emag.Systems;
 using Content.Shared.Wires;
 
 namespace Content.Server.Access;
 
 public sealed class AccessWireAction : ComponentWireAction<AccessReaderComponent>
 {
+    [Dependency] private readonly EmagSystem _emagSystem = default!;
```
Comment (deltanedas) [inline Content.Server/Access/AccessWireAction.cs:12]: i have no idea if adding systems to an action is evil

**Comment by deltanedas** [comment]
@EmoGarbage404 are above concerns valid or is it ok

**Comment by EmoGarbage404** [comment]
> @EmoGarbage404 are above concerns valid or is it ok

protected system is good.
you can't have a dependency in an action, so just do EntityManager.System<T>() to get the system when needed.

**Comment by deltanedas** [comment]
ok i think i broke it gonna remake

**Comment by deltanedas** [comment]
remade in #14096

**Pull request closed without merging**